lib/internal/fs/streams.js

```diff
@@ -67,6 +72,26 @@ function ReadStream(path, options) {
   // For backwards compat do not emit close on destroy.
   options.emitClose = false;
 
+  if (options.open && typeof options.open !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.open', 'function', options.open);
+  }
+
+  if (options.read && typeof options.read !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.read', 'function', options.read);
+  }
+
+  if (options.close && typeof options.close !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE('options.close', 'function', options.close);
+  }
+
+  this[kOpen] = options.open;
+  this[kRead] = options.read;
+  this[kClose] = options.close;
+
+  delete options.open;
+  delete options.read;
+  delete options.close;
```

Comment:
Using a an object wrapper as discussed _before_ would actually avoid these bunch of deletes, which in turn would ensure that we keep the object with "fast properties" access. I'd actually prefer that. Maybe we can call it "overrides" or "internal"?

cc @jasnell

Reply:
If they need to be deleted at all, setting to undefined rather than deleting would avoid the performance penalty. Why do they need to be removed?

Reply:
`read` is a known option of streams.

Reply:
@jasnell: Because the base stream implementation will use them to override as well. See, https://github.com/nodejs/node/blob/master/lib/_stream_writable.js#L226.

Reply:
Ah right. Then just setting them to undefined would avoid that issue.

Reply:
Fixed... let me know what the consensus will be in regards to fs object or not

Reply:
I'm a little worried that this way might need a semvar-major... in case someone wants to override the methods of `Writable`? Seems very unlikely though.